src/arrayMethodSort.js

```diff
@@ -4,8 +4,24 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let comparator = compareFunction;
+
+    if (typeof comparator !== 'function') {
+      comparator = function (a, b) {
+        return String(a) > String(b) ? 1 : -1;
```

Comment:
The default comparator should return 0 when elements are equal to ensure stability in sorting. Also, the comparison should handle the case where elements are equal to each other, not just when one is greater than the other.